Plugin Recaptcha and Mootools on Joomla 3.4

Qlimax90 opened the pull request:
Hi everyone,

in /plugins/captcha/recaptcha/recaptcha.php from line 55 to 79, is there any reason why the new recaptcha is loading the mootools library? Especially on line 76 Mootools gets loaded by JHtml::_('script', $file, true, true);

https://github.com/joomla/joomla-cms/issues/6212

Qlimax90 committed:
Plugin Recaptcha and Mootools on Joomla 3.4

Hi everyone,

in /plugins/captcha/recaptcha/recaptcha.php from line 55 to 79, is there any reason why the new recaptcha is loading the mootools library? Especially on line 76 Mootools gets loaded by JHtml::_('script', $file, true, true);

https://github.com/joomla/joomla-cms/issues/6212

mbabker commented on plugins/captcha/recaptcha/recaptcha.php:
```diff
@@ -59,8 +59,6 @@ public function onInit($id = 'dynamic_recaptcha_1')
 
 				$file = $app->isSSLConnection() ? 'https' : 'http';
 				$file .= '://www.google.com/recaptcha/api/js/recaptcha_ajax.js';
-				JHtml::_('script', $file);
```
Removing this line completely means you aren't loading the recaptcha_ajax.js file.  FWIW this call isn't loading MooTools.

dgt41 replied:
```
JHtml::_('script', $file, false, true);
```
This code will not load mootools and also keeps the loading chain untouched, also see the comment below

dgt41 replied:
Also using `addScript()` instead of `JHtml::_()` disables the overriding feature of the script file

losedk commented:
Tested. All good

C-Lodder commented:
The `addScriptDeclaration` is being injected before the `addScript` statement so would this not lead to undefined variables?

losedk commented:
Is that a problem? addScript always injects into head

dgt41 commented:
@C-Lodder It would be nice to follow the actual chain of events, first the file and then the initialization :+1:

Qlimax90 commented:
hmm but this would cause an redundancy in the code :-1:

dgt41 commented on plugins/captcha/recaptcha/recaptcha.php:
```diff
@@ -72,15 +70,13 @@ public function onInit($id = 'dynamic_recaptcha_1')
 				$file .= '://www.google.com/recaptcha/api.js?hl=' . JFactory::getLanguage()
 						->getTag() . '&onload=onloadCallback&render=explicit';
-
-				JHtml::_('script', $file, true, true);
```
```
JHtml::_('script', $file, false, true);
```

wilsonge replied:
We don't want relative file paths - this is the google script - which isn't stored in Joomla ;)

dgt41 replied:
That’s my bad! Of course there is no way to override an external file 😜

Qlimax90 committed:
Update recaptcha.php

This should not disable overwriting, doesn't load mootools and also does not provocate code redundancy.
What about the loading chain?

wilsonge commented:
@C-Lodder The inline scripts are always added after the files so it's not an issue. That's why we load jQuery no captcha as a file rather than a script

nueckman commented:
Looks good

anibalsanchez commented:
@test It works Ok, no mootools.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6218">issues.joomla.org/joomla-cms/6218</a>.</sub>

zero-24 commented:
@dgt41 @Qlimax90  what is the status here? Can we leave it now `as it is` or need we to change something? As we get multible successful tests here :smile: Thanks.

Qlimax90 commented:
I think we can leave it as it is :)

dgt41 commented:
@Qlimax90 just do a merge on staging to resolve the conflicts

Qlimax90 commented:
@dgt41 It seems that i have not the rights to do this?
"This pull request contains merge conflicts that must be resolved.
Only those with write access to this repository can merge pull requests."

dgt41 commented:
@Qlimax90 if you pull rebase to latest staging in your local copy, solve the conflicts and push the code back to your online copy, everything should be ok!

Kubik-Rubik commented:
@Qlimax90 Will you update your PR or should I recreate it? I think that it is a good change (get rid of the loading of the Mootools framework). If you are still keen to do it, then please do it in the next days if possible! If you do it, then please also add a blank line after the switch statement. Thank you in advance!

Qlimax90 commented:
@dgt41 @Kubik-Rubik Please feel free to recreate, at the moment I am struggeling with the local copy due to the fact that I created the PR online. The local branch is not working, sorry...

Kubik-Rubik commented:
Closed in favor of #7403. Thank you @Qlimax90 for your contribution!
